OpenSim/Wrapping/Java/swig/javaWrapOpenSim.i

```diff
@@ -565,6 +565,41 @@ SWIG_JAVABODY_PROXY(public, public, SWIGTYPE)
 	}
 };
 
+/*
+Extend concrete Joints to use the inherited base constructors.
+Note that CustomJoint and EllipsoidJoint do implement their own
+constructors because they have additional arguments.
+*/
+%define EXPOSE_JOINT_CONSTRUCTORS_HELPER(NAME)
+%extend OpenSim::NAME {
+	NAME(const std::string& name,
+         const std::string& parentName,
+         const std::string& childName) {
+		return new NAME(name, parentName, childName, false);
+	}
+	
+	NAME(const std::string& name,
+         const PhysicalFrame& parent,
+         const SimTK::Vec3& locationInParent,
+         const SimTK::Vec3& orientationInParent,
+         const PhysicalFrame& child,
+         const SimTK::Vec3& locationInChild,
+         const SimTK::Vec3& orientationInChild) {
+		return new NAME(name, parent, locationInParent, orientationInParent,
+					child, locationInChild, orientationInChild, false);
+	}
+};
+%enddef
+
+EXPOSE_JOINT_CONSTRUCTORS_HELPER(FreeJoint);
+EXPOSE_JOINT_CONSTRUCTORS_HELPER(BallJoint);
+EXPOSE_JOINT_CONSTRUCTORS_HELPER(PinJoint);
+EXPOSE_JOINT_CONSTRUCTORS_HELPER(SliderJoint);
+EXPOSE_JOINT_CONSTRUCTORS_HELPER(WeldJoint);
+EXPOSE_JOINT_CONSTRUCTORS_HELPER(GimbalJoint);
+EXPOSE_JOINT_CONSTRUCTORS_HELPER(UniversalJoint);
+EXPOSE_JOINT_CONSTRUCTORS_HELPER(PlanarJoint);
+
 
```

Comment:
This swig macro implements the previously-existing constructor and also provides the new Joint constructor. In python, we do have tests for the previously-existing constructor but not for new one.

Might be good to add comments in the swig .i file saying that these are only here b/c swig doesn't generated constructors for C++11 constructor inheritance (`using Derived::Derived;`).